### ISSUE TITLE
Kubernetes core-dns and calico-kube-controllers pods do not start up because calico plugin cannot be found

### ISSUE THREAD

**Issue by Placeblock** [projectcalico/calico#9336]
I set up a Kuberenetes Cluster using Kubeadm and after that I installed Calico for self-managed on premise deployments (https://docs.tigera.io/calico/latest/getting-started/kubernetes/self-managed-onprem/onpremises)

## Expected Behavior

The cluster should start up and use calico as the networking plugin.

## Current Behavior

The coredns and calico-kube-controllers pods don't start up and stay in the "ContainerCreating" state.
The following errors show up in the /var/log/kubelet/kubelet.log file:
```
E1013 14:57:10.206107    2386 kuberuntime_manager.go:1479] "Failed to stop sandbox" podSandboxID={"Type":"containerd","ID":"8697dcd4338aa6452c88508e3b46888244111ac05d8d569e59dcc4062b4e8775"}
E1013 14:57:10.206130    2386 kuberuntime_manager.go:1079] "killPodWithSyncResult failed" err="failed to \"KillPodSandbox\" for \"3b02d395-2e06-4bcb-946d-f67ab3f1edf3\" with KillPodSandboxError: \"rpc error: code = Unknown desc = failed to destroy network for sandbox \\\"8697dcd4338aa6452c88508e3b46888244111ac05d8d569e59dcc4062b4e8775\\\": plugin type=\\\"calico\\\" failed (delete): failed to find plugin \\\"calico\\\" in path [/usr/libexec/cni]\""
E1013 14:57:10.206146    2386 pod_workers.go:1301] "Error syncing pod, skipping" err="failed to \"KillPodSandbox\" for \"3b02d395-2e06-4bcb-946d-f67ab3f1edf3\" with KillPodSandboxError: \"rpc error: code = Unknown desc = failed to destroy network for sandbox \\\"8697dcd4338aa6452c88508e3b46888244111ac05d8d569e59dcc4062b4e8775\\\": plugin type=\\\"calico\\\" failed (delete): failed to find plugin \\\"calico\\\" in path [/usr/libexec/cni]\"" pod="calico-system/csi-node-driver-khvcm" podUID="3b02d395-2e06-4bcb-946d-f67ab3f1edf3"
```

## Possible Solution

Somehow it tries to find the calico plugin inside the /usr/libexec/cni folder, but it is located inside /opt/cni/bin.
Do I have to reconfigure anything?

## Steps to Reproduce (for bugs)

1. Setup a Kubernetes cluster using kubeadm
2. Install calico for self-managed on premise deployments

## Context

I'm trying to setup a kubernetes cluster using calico for network management

## Your Environment

* Calico version: 3.28.2
* Calico dataplane (iptables, windows etc.): iptables
* Orchestrator version (e.g. kubernetes, mesos, rkt): kubernetes
* Operating System and version: Alpine 3.13.12
* Kubectl, Kubeadm and Kubelet packages where pulled from the Alpine edge branch, as they where not included in the 3.13 branch

**Comment by Placeblock** [comment]
I just tested it with Alpine 3.20 and it does still not work

**Comment by Placeblock** [comment]
Copying the calico and calico-ipam plugins from the opt folder to the /usr/libexec/cni solves all issues and the pods start just fine. 

However, I still think that this is not as it should be. If I understood correctly it should "just work"? Any Ideas what could have went wrong in the installation? I already had a hart time finding anything cni - kubernetes related and I did not find anything about my Issue.

**Comment by MichalFupso** [comment]
Hi @Placeblock, what container runtime are you using? You might have to make changes to the default CNI binary path for your container runtime.

**Comment by caseydavenport** [comment]
> Copying the calico and calico-ipam plugins from the /opt/cni/bin directory to the /usr/libexec/cni directory solves all issues and the pods start just fine.

Yep, sounds likely that your cluster configuration is just looking in a different place for the binaries than Calico is configured to place them. Different platforms / installers have different CNI configurations.

How are you installing Calico? the tigera/operator has some concept of platform, so is able to modify this path based on the installer (to a degree).

If you're installing with `calico.yaml` then you need to make sure your install path in the manifest matches the location your cluster expects CNI plugins to be installed.
